src/models/BaseModel.py

```diff
@@ -1,118 +1,205 @@
-        self.train_source_name_feature = self.train_data[['SourceName']]
-        self.train_location_raw_feature = self.train_data[['LocationRaw']]
-        self.train_salary_normalized = self.train_data[['SalaryNormalized']]
-
-        self.test_description_feature = self.test_data[['FullDescription']]
-        self.test_salary_normalized = self.test_data[['SalaryNormalized']]
-
-        # can be disabled to speed up load times
-        if load_location:
-            if BaseModel.normalized_location_data is None:
-                self.load_location_data(train_normalized_location_file_name)
-            train_normalized_location_data = \
-                BaseModel.normalized_location_data[:self.training_data_size]
-
-            # test_normalized_location_data = normalized_location_data[-self.test_data_size:]
-
-            self.cleaned_town_feature = train_normalized_location_data[['town']]
-            self.cleaned_region_feature = train_normalized_location_data[['region']]
+    TRAIN_NORMALIZED_LOCATION_FILE_NAME = '../data/train_normalised_location.csv'
+    TRAIN_DATA_CSV_FILE_NAME = '../data/Train_rev1.csv'
+    CLEANED_FILE_NAME = '../data/Binary_Preprocessed_Data.csv'
+
+    def __init__(self, train_size=0.75):
+        print('Initialized BaseModel')
+
+        if os.path.exists(BaseModel.CLEANED_FILE_NAME):
+            print('Pre-processed data exists, reading from the file')
+            self.cleaned_encoded_data = self.load_cleaned_data()
+            print('Data read complete')
+        else:
+            print('Pre-processed data doesn\'t exist, preprocessing data first')
+            print('This operation will take a while')
+            self.load_all_data()
+            print('Raw data read complete')
+            # Get features to clean and reshape
+            self.description_feature = BaseModel.data[['FullDescription']]
+            self.contract_type_feature = BaseModel.data[['ContractType']]
+            self.contract_time_feature = BaseModel.data[['ContractTime']]
+            self.category_feature = BaseModel.data[['Category']]
+            self.company_feature = BaseModel.data[['Company']]
+            self.source_name_feature = BaseModel.data[['SourceName']]
+            self.location_raw_feature = BaseModel.data[['LocationRaw']]
+            self.title_feature = BaseModel.data[['Title']]
+            self.salary_feature = BaseModel.data[['SalaryNormalized']]
+            # Read cleaned locations from separate file, cleaned with Google location
+            self.cleaned_town_feature = BaseModel.normalized_location_data[['town']]
+            self.cleaned_region_feature = BaseModel.normalized_location_data[['region']]
+
+            (self.processed_data, self.feature_list) = self.preprocess_data()
+            self.export_data(self.processed_data, 'Binary_Preprocessed_Data')
+            self.export_data(self.feature_list, 'Feature_List')
+            self.cleaned_encoded_data = self.load_cleaned_data()
```

Comment:
my reasoning for that is, even if you clean the data or not clean the data, read it with a single function (load_cleaned_data()), ( it takes like half a second ) with doing that you know if there was a problem right after cleaning since you load it from the exported file